packages/core/src/lib/history.ts

```diff
@@ -0,0 +1,110 @@
+import { LogResult, simpleGit } from 'simple-git';
+import {
+  CoreConfig,
+  Format,
+  PersistConfig,
+  UploadConfig,
+  uploadConfigSchema,
+} from '@code-pushup/models';
+import { getCurrentBranchOrTag, safeCheckout } from '@code-pushup/utils';
+import { collectAndPersistReports } from './collect-and-persist';
+import { GlobalOptions } from './types';
+import { upload as uploadCommandLogic } from './upload';
+
+export type HistoryOnlyOptions = {
+  targetBranch?: string;
+  uploadReports?: boolean;
+  forceCleanStatus?: true;
+};
+export type HistoryOptions = Required<
+  Pick<CoreConfig, 'plugins' | 'categories'> & {
+    persist: Required<PersistConfig>;
+  } & { upload: Required<UploadConfig> }
+> &
+  GlobalOptions &
+  HistoryOnlyOptions;
+
+export async function history(
+  config: HistoryOptions,
+  commits: string[],
+): Promise<string[]> {
+  const initialBranch: string = await getCurrentBranchOrTag();
+
+  const { uploadReports = true } = config as unknown as HistoryOptions;
```

Comment:
I usually define optional boolean parameters so it's off by default, then I can treat all falsy values the same. You can always invert the parameter meaning, e.g. this would become `skipUploads`.

Reply:
good input! Changed it already!